From 8a9a2f517e9ed21e39883e52b7a989dbcdccb968 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 20 Nov 2023 10:25:36 +0100
Subject: [PATCH 0355/1695] drivers: stm32_tamp: use clk_enable() error code

Use error code returned by clk_enable() at probe.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 73c1e2edcc67a71ba1a24a2fcb8058a5d1e6bc15 drivers: stm32_tamp: use clk_enable() error code]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: If403081ca02ba1a5413f79823a927f42795518f2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/343119
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_tamp.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index 9b9fae0ec..83b86709d 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -1741,7 +1741,9 @@ static TEE_Result stm32_tamp_probe(const void *fdt, int node,
 		return res;
 
 	/* Init Tamp clock */
-	clk_enable(stm32_tamp.pdata.clock);
+	res = clk_enable(stm32_tamp.pdata.clock);
+	if (res)
+		return res;
 
 	base = io_pa_or_va(&stm32_tamp.pdata.base, 1);
 
-- 
2.47.2

